telegram/ext/_jobqueue.py

```diff
@@ -821,8 +842,25 @@ def next_t(self) -> Optional[datetime.datetime]:
         return self.job.next_run_time
 
     @classmethod
-    def _from_aps_job(cls, job: "APSJob") -> "Job[CCT]":
-        return job.func.__self__
+    def from_aps_job(cls, job: "APSJob") -> "Job[CCT]":
+        """Provides the :class:`telegram.ext.Job` that is associated with the given APScheduler
+        job.
+
+        Tip:
+            This method can be useful when using advanced APScheduler features along with
+            :class:`telegram.ext.JobQueue`.
+
+        .. versionadded:: NEXT.VERSION
+
+        Args:
+            job (:class:`apscheduler.job.Job`): The APScheduler job
+
+        Returns:
+            :class:`telegram.ext.Job`
+        """
+        job_ = job.args[1]
+        job_._job = job  # pylint: disable=protected-access
+        return job_
```

Comment:
csi comments / rename variables here? `job`, `_job`, `job_` need to be different from each other haha

Reply:
Perhaps changing the argument name `job` ->  `aps_job` will be better